### ISSUE TITLE
[BUG] Gateway's version resolving need to be improved

### ISSUE THREAD

**Issue by devaskim** [thingsboard/thingsboard-gateway#984]
**Describe the bug**
As of 3.2 version resolving is based on distribution information, but there is Docker deployment and manual/IDE running option where is no distribution.

**Error traceback (If available):**
```
[===UPDATE===]

 New version 3.1 is available!

[===UPDATE===]
"
""2022-11-09 10:49:48" - |INFO| - [tb_updater.py] - tb_updater - check_for_new_version - 84 - 3.1"
""2022-11-09 10:49:48" - |INFO| - [tb_updater.py] - tb_updater - check_for_new_version - 86 -

[===UPDATE===]

 New version 3.1 is available!

[===UPDATE===]

```

**Versions (please complete the following information):**
 - Thingsboard IoT Gateway version 3.2

**Comment by devaskim** [comment]
@imbeacon The issue reproduces again. Seems like the following changes were reverted.
![Скриншот 10-12-2023 142258](https://github.com/thingsboard/thingsboard-gateway/assets/86909129/445d70e5-4cc9-443c-aa16-d1406f91e08c)

**Comment by devaskim** [comment]
@imbeacon Please, reopen because of [reverting this pr](https://github.com/thingsboard/thingsboard-gateway/commit/65e756786adb209c7c818589a0c3ae67e6181347) it reproduces again.

**Comment by imbeacon** [comment]
@devaskim, yes, it was reverted, because it causes an issue with remote configuration.

**Comment by devaskim** [comment]
@imbeacon ok, but what about the original issue that I mentioned in the description?

**Comment by devaskim** [comment]
@imbeacon Please, take a look at https://github.com/thingsboard/thingsboard-gateway/pull/1317
